lib/tom_queue/deferred_work_manager.rb

```diff
@@ -87,7 +86,7 @@ def start
       # (which have been scheduled by the AMQP consumer). If a message is returned
       # then we re-publish the messages to our internal QueueManager and ack the deferred
       # message
-      until @shutdown
+      loop do
         # This will block until work is ready to be returned, interrupt
```

Comment:
The reason for not removing interrupt is that it's still interrupted by `SignalException`.